common/primitives/src/messages.rs

```diff
@@ -15,9 +15,10 @@ use utils::*;
 pub struct MessageResponse<AccountId, BlockNumber> {
 	#[cfg_attr(feature = "std", serde(with = "as_hex"))]
 	pub data: Vec<u8>, //  Serialized data in a user-defined schema format
-	pub signer: AccountId,       //  Signature of the signer
-	pub msa_id: MessageSenderId, //  Message source account id (the original sender)
-	pub index: u16,              // index in block to get total order
+	pub signer: AccountId,                     //  Signature of the signer
+	pub sender_msa_id: MessageSenderId,        //  Message source account id (the original sender)
+	pub on_behalf_of: Option<MessageSenderId>, //  Message source account id (the original message producer)
+	pub index: u16,                            // index in block to get total order
 	pub block_number: BlockNumber,
 }
 
```

Comment:
@aramikm  what are your thoughts, given messages can be added by sender msa_id but sender could be a provider submitting on behalf of a delegator

So is it a goo update to message being stored?

Reply:
I don't think `on_behalf_of` is necessary since we can get msa_id of the signer via stored `signer` key.